### PR TITLE
TYP: add some types to pandas/io/parquet.py

### PR DESCRIPTION
**Pull request by simonjayhawkins**
broken off #28339

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
lgtm

File: pandas/io/parquet.py [hunk 1]
```diff
@@ -35,19 +36,15 @@ def get_engine(engine):
...
     if engine == "pyarrow":
         return PyArrowImpl()
     elif engine == "fastparquet":
         return FastParquetImpl()
 
+    raise ValueError("engine must be one of 'pyarrow', 'fastparquet'")
 
-class BaseImpl:
-
-    api = None  # module
```
Comment (jbrockmendel) [inline pandas/io/parquet.py]: this is just unused?

**Comment by jbrockmendel** [comment]
small question, otherwise LGTM

**Comment by jreback** [comment]
thanks @simonjayhawkins

File: pandas/io/parquet.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/io/parquet.py]: it doesn't appear to be necessary to be declared here. not used by the BaseImpl methods so not technically a mandatory property.

mypy errors were
```
pandas\io\parquet.py:101: error: "None" has no attribute "Table"
pandas\io\parquet.py:103: error: "None" has no attribute "parquet"
pandas\io\parquet.py:112: error: "None" has no attribute "parquet"
```

**Pull request merged**